widgetastic_manageiq/__init__.py

```diff
@@ -4207,3 +4207,153 @@ def data_for_timestamp(self, timestamp):
         for row in self.tooltip.rows():
             tooltip_data.update({row[0].text: row[1].text})
         return tooltip_data
+
+# Widget and its form views for Infra Planning Wizard
+
+
+class MultiSelectList(Widget):
+    """Multi Select Box to be used in Infra Planning Wizard for v2v.
+
+    Args:
+        div_id (str): id of div element that contains multiselectbox
+
+    Usage:
+        e.g. target_clusters = MultiSelectList('target_clusters')
+    """
+    ROOT = ParametrizedLocator('.//div[contains(@id,{@div_id|quote})]')
+    ITEMS_LOCATOR = './div/div/div/span[contains(@class,"dual-pane-mapper-item-container")]'
+    SELECTED_ITEMS_LOCATOR = ('./div[contains(@class,"selected")]/div/div'
+    '/span[contains(@class,"dual-pane-mapper-item-container")]')
+
+    def __init__(self, parent, div_id, logger=None):
+        Widget.__init__(self, parent, logger=logger)
+        self.div_id = div_id
+
+    @property
+    def all_items(self):
+        """Returns list of all items."""
+        return [self.browser.text(item) for item in self.all_items_locators]
+
+    @property
+    def item_count(self):
+        """Returns item count."""
+        return len(self.all_items)
+
+    @property
+    def list_selected_items(self):
+        """Returns list of items that are currently selected."""
+        return [self.browser.text(item) for item in self.list_selected_items_locators]
+
+    @property
+    def all_items_locators(self):
+        """Returns list of item locators for each of the items."""
+        return self.browser.elements(self.ITEMS_LOCATOR)
+
+    @property
+    def list_selected_items_locators(self):
+        """Returns list of item locators for currently selected items."""
+        return self.browser.elements(self.SELECTED_ITEMS_LOCATOR)
+
+    def _get_all_options(self):
+        """ Method to return dict of all the item webelement locators and text"""
+        return {self.browser.text(element): element for element in self.all_items_locators}
+
+    def read(self):
+        """Method to read elements that are currently selected."""
+        return self.list_selected_items
+
+    def fill(self, values, partial_match=False):
+        """Click/Select multiple the items in the multiselectbox
+
+        Args:
+            items: 'list' of items of type 'str' to be matched
+            partial_match: if you want to use partial text matching. Default:False
+
+        Returns:
+            True if all values were found and clicked.
+        """
+        # TODO self.browser.plugin.ensure_page_safe() should be here
+        wait_for(lambda: "loading" not in self.browser.classes(self), timeout=5)
+        if values == self.list_selected_items or values == []:
+            return False
+        if values:
```

Comment:
I don't think this logic is right, as if `values` is None, this fill method will return True without doing anything.

I think the if statement above should also check `or values is None`, and then there's no need for this additional if statement.